restapi/endpoints_test.go

```diff
@@ -0,0 +1,146 @@
+package restapi
+
+import (
+	"fmt"
+	"net/http"
+	"github.com/spf13/viper"
+	"net/http/httptest"
+	"testing"
+	"github.com/gavv/httpexpect/v2"
+	"github.com/stretchr/testify/suite"
+	"github.com/uwblueprint/shoe-project/internal/database/migrations"
+	"github.com/uwblueprint/shoe-project/internal/database/models"
+	"github.com/uwblueprint/shoe-project/testutils"
+	"gorm.io/gorm"
+)
+
+type endpointTestSuite struct {
+	suite.Suite
+	endpoint *httpexpect.Expect
+	db       *gorm.DB
+	token     string
+}
+
+func (suite *endpointTestSuite) SetupSuite() {
+	db, err := testutils.MockDatabase()
+	if err != nil {
+		suite.Fail("error while creating database", err)
+	}
+	suite.db = db
+
+	// setup required for jwt authentication
+	viper.SetDefault("auth.jwt_key", "GotFKl1PGgMpLg7D36NiI0hy/gsl6woTCXYdhKATbzc=")
```

Comment:
please make sure that our secrets env vars are different in the `.env` file. feel free to generate new tokens, users, passwords etc and send us the updated `.env` file.

Reply:
I don't think it actually matters what I set the` auth.jwt_key` here? Setting it as `	viper.SetDefault("auth.jwt_key", "random")
` still passes all the tests? 😕

Reply:
yes, that's the way it should be in all passing tests (because we are making sure to send the correct token in the header for all our post requests).

actually, this makes me think we should have failing tests for login (i.e. username or pw is incorrect), and tests for unauthorized POST requests (where the token is missing or incorrect). _but_, because our app is quite small, we maybe can get away with testing only happy paths. i don't have a strong opinion, maybe others can comment

Reply:
we should have a different suite all together for login testing

Reply:
@s278gupt we discussed in the wed meeting to put login testing and get countries testing as a separate ticket ^^